assets/scripts/streets/data_model.js

```diff
@@ -441,3 +442,33 @@ export function updateEverything (dontScroll, save = true) {
     scheduleSavingStreetToServer()
   }
 }
+export function createEmptyStreet (state) {
+  const { units } = state.persistSettings
+  const { signedIn, signInData } = state.user
+  const { newStreetPreference } = state.settings
+  const createDefaultStreet = newStreetPreference !== NEW_STREET_EMPTY
+  const segments = createDefaultStreet ? fillDefaultSegments(units) : []
+  const leftBuildingHeight = createDefaultStreet ? DEFAULT_BUILDING_HEIGHT_LEFT : DEFAULT_BUILDING_HEIGHT_EMPTY
+  const rightBuildingHeight = createDefaultStreet ? DEFAULT_BUILDING_HEIGHT_RIGHT : DEFAULT_BUILDING_HEIGHT_EMPTY
+
+  const leftBuildingVariant = createDefaultStreet ? DEFAULT_BUILDING_VARIANT_LEFT : DEFAULT_BUILDING_VARIANT_EMPTY
+  const rightBuildingVariant = createDefaultStreet ? DEFAULT_BUILDING_VARIANT_LEFT : DEFAULT_BUILDING_VARIANT_EMPTY
+
+  return {
+    units: units,
+    location: null,
+    name: null,
+    userUpdated: false,
+    editCount: 0,
+    width: normalizeStreetWidth(DEFAULT_STREET_WIDTH, units),
+    environment: DEFAULT_ENVIRONS,
+    leftBuildingHeight,
+    leftBuildingVariant,
+    rightBuildingHeight,
+    rightBuildingVariant,
+    schemaVersion: LATEST_SCHEMA_VERSION,
+    segments,
+    updatedAt: new Date().toISOString(),
+    creatorId: (signedIn && signInData.userId) || null
+  }
```

Comment:
It looks like some of this duplicates `prepareDefaultStreet()`.

At some point I refactored `fillEmptySegments()` to be a pure(r) function. It might be a good idea to refactor `prepareDefaultStreet()` and `prepareEmptyStreet()` to return street data (or create a new function to do so) that can be used for both the original and revised API.